### PR TITLE
reapplying system theme when app resumes

### PR DESCRIPTION
**Pull request by abdul10ahad**
Reapply system theme to solve the status bar color issue whenever the app resumes

### CODE REVIEW

File: packages/stacked_themes/lib/src/theme_builder.dart [hunk 1]
```diff
@@ -58,4 +60,62 @@ class _ThemeBuilderState extends State<ThemeBuilder> {
...
+    WidgetsBinding.instance.removeObserver(this);
+  }
+
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    super.didChangeAppLifecycleState(state);
+
+    switch (state) {
+      case AppLifecycleState.inactive:
+        print('appLifeCycleState inactive');
```
Comment (FilledStacks) [inline packages/stacked_themes/lib/src/theme_builder.dart]: Please remove the print statements or make it optional

File: packages/stacked_themes/lib/src/theme_builder.dart [hunk 2]
```diff
@@ -58,4 +60,62 @@ class _ThemeBuilderState extends State<ThemeBuilder> {
...
+  }
+
+  //NOTE: re-apply the appropriate theme when the application gets back into the foreground
+  void adjustSystemThemeIfNecessary() {
+    switch (themeManager.selectedThemeMode) {
+      //do nothing
+      case ThemeMode.dark:
+        break;
+      //do nothing
+      case ThemeMode.dark:
```
Comment (FilledStacks) [inline packages/stacked_themes/lib/src/theme_builder.dart:107]: 2 cases for `ThemeMode.dark` one of them has to be `ThemeMode.light`

File: packages/stacked_themes/lib/src/theme_builder.dart [hunk 3]
```diff
@@ -58,4 +60,62 @@ class _ThemeBuilderState extends State<ThemeBuilder> {
...
+        break;
+      //reapply theme
+      case ThemeMode.system:
+        print('reApplying system theme');
```
Comment (FilledStacks) [inline packages/stacked_themes/lib/src/theme_builder.dart]: Please remove the log or make it optional.

**Review by FilledStacks — CHANGES_REQUESTED** [review]
Looks good, just the logs that I would prefer to be optional. The way that optional logs are usually implemented is by supplying a verbose bool flag to the object printing it out. The bool should be false be default and the logs should only print if it's true. 

Another thing to do is format the logs properly. To avoid cluttering the user of the packages logs you should change the format to the following. 

NameOfClass - Function Name - Log Message

This will help the user determine where the log is coming from and make it much easier to search for when looking for bugs. In our case the logs would be 

```
print('ThemeManager - didChangeAppLifecycleState - appLifeCycleState resumed')
```

File: packages/stacked_themes/lib/src/theme_builder.dart (hunk 3 above)
Comment (abdul10ahad) [reply packages/stacked_themes/lib/src/theme_builder.dart]: thanks for the feedback I'll update the pull request

**Comment by abdul10ahad** [comment]
I've made the changes required

**Comment by FilledStacks** [comment]
Thanks @frodo10messi  will bump the version, test it out and then push if it's ready.

**Pull request merged**